app/app.go

```diff
@@ -253,54 +259,37 @@ func (app *App) initServices(ctx context.Context) error {
 	app.setStatus(statusInitializing)
 	app.logger.Infof("initialize services...")
 
-	wg := &sync.WaitGroup{}
-	errors := make(chan error, len(app.services))
+	var err error
 
-	wg.Add(len(app.services))
 	for _, svc := range app.services {
-		go func(svc interface{}) {
-			defer wg.Done()
-
-			// Register checks before starting initialization
-			if check, ok := svc.(Checkable); ok {
-				if err := check.RegisterCheck(app.readiness); err != nil {
-					errors <- err
-					return
-				}
+		// Register checks before starting initialization
+		if check, ok := svc.(Checkable); ok {
+			if err = check.RegisterCheck(app.readiness); err != nil {
+				break
 			}
+		}
 
-			// Register metrics before starting initialization
-			if measure, ok := svc.(Measurable); ok {
-				if err := measure.RegisterMetrics(app.prometheus); err != nil {
-					errors <- err
-					return
-				}
+		// Register metrics before starting initialization
+		if measure, ok := svc.(Measurable); ok {
```

Comment:
I think it is fine to `RegisterMetrics` before `Init(...)`

Reply:
Same as previous commnet. Let's change method order in another PR.